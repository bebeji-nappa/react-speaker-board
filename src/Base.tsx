import React, { useState, useEffect } from "react";
import "./types"
import { Control } from "./Control";

const modeValue = [
  "slide",
  "inlineSlide"
]

export const ReactSpeakerBoard:React.VFC<SpeakerBoardProps> = ({ mode = "slide", slide, controlTheme = "#71a3e3", controlBar = "none", width = "80vw", height = "80vh" }) => {
  const [view, setView] = useState<JSX.Element>(slide[0]);
  const [index, setIndex] = useState<Number>(0);
  const max = slide.length - 1;

  const NextSlide = () => {
    if (index !== max) {
      let num = Number(index)
      setIndex(Number(num + 1))
    }
  }
  
  const ProvSlide = () => {
    if (index !== 0) {
      let num = Number(index)
      setIndex(Number(num - 1))
    }
  }

  useEffect(() => {
    setView(slide[Number(index)])
  }, [index])

  return (
    <>
     {modeValue.includes(mode) && 
      mode === "inlineSlide" ? 
      <div className={mode} style={{ width: `${width}`, height: `${height}` }}>
        {view}
        <Control next={() => NextSlide()} prov={() => ProvSlide()} color={controlTheme} controlBar={controlBar} />
      </div> :
      <div className={mode}>
        {view}
        <Control next={() => NextSlide()} prov={() => ProvSlide()} color={controlTheme} controlBar={controlBar} />
      </div>
     }
    </>
  )
}
