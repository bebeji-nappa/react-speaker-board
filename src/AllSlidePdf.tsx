import React, { useEffect, useCallback } from "react";
import { useSlideOutputFiles } from "./index";

type AllSlideProps = {
  slide: (() => JSX.Element)[];
};

const AllSlidePdf: React.FC<AllSlideProps> = ({ slide }) => {
  const { download } = useSlideOutputFiles();

  const downloadPdf = useCallback(() => {
    download(slide, "reactspeakerboard_test.pdf");
  }, []);

  return (
    <div style={{ display: "flex", flexDirection: "column", alignItems: "center", width: "100%", height: "auto", overflow: "scroll", gap: "20px" }}>
      <button onClick={downloadPdf} style={{ margin: "20px 0", cursor: "pointer", fontSize: "16px" }}>
        PDF Download
      </button>
      {slide.map((Slide, index) => {
        return (
          <div id={`slide-${index+1}`} key={index} style={{ width: "100vw", height: "100vh", aspectRatio: "16 / 9" }}>
            <Slide />
          </div>
        );
      })}
    </div>
  )
};

export default AllSlidePdf;
