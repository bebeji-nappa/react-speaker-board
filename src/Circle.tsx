import * as React from "react";
import "./types"

const Circle: React.VFC<FigureProps> = ({ children, backgroundColor = "white", color = "black", width = "100px", height = "100px" }) => {
  return (
    <>
      {children ? 
        <div className="cicle" style={{ backgroundColor: backgroundColor, color: color, width: width, height: height }}>
          {children}
        </div> :
        <div className="circle" style={{ backgroundColor: backgroundColor, color: color, width: width, height: height }}></div>
      }
    </>
  )
}

export default Circle
