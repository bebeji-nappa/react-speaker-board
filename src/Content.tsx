import * as React from "react";
import "./types"

const alignValue = [
  "left",
  "center",
  "right"
]

const sizeValue = [
  "xsmall",
  "small",
  "medium",
  "large",
  "xlarge",
  "xxlarge",
  "xxxlarge",
]

const Content:React.VFC<ContentProps> = ({children, align = "left", size = "medium"}) => {
  return (
    <div className={`Content${alignValue.includes(align) && align ? ` ${align}` : ""}${sizeValue.includes(size) && size ? ` size-${size}` : ""}`}>{children}</div>
  )
}

export default Content
