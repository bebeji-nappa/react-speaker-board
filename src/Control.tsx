import React from "react";
import "../styles/Control.css"

export const Control: React.VFC<Control> = ({ next, prov, color, controlBar }) => {
  return (
    <div className={`Control-wrap background-${controlBar}`}>
      <div className="Control">
        <button onClick={prov}>
          <svg width="25px" height="25px" viewBox="0 0 291 338" fill="none" xmlns="http://www.w3.org/2000/svg" className="slide-button">
          <path d="M0 169L291 0.991089V337.009L0 169Z" fill={color}/>
          </svg>
        </button>
        <button onClick={next}>
          <svg width="25px" height="25px" viewBox="0 0 291 338" fill="none" xmlns="http://www.w3.org/2000/svg" className="slide-button">
          <path d="M291 169L0 337.009V0.991074L291 169Z" fill={color}/>
          </svg>
        </button>
      </div>
    </div>
  )
}
