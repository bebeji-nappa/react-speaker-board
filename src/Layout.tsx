import React from "react"
import "../styles/Layout.css"

export const Layout: React.VFC<LayoutProps> = ({ children, layout }) => {
  return (
    <div className={`board ${layout}`}>
      {children}
    </div>
  )
}
