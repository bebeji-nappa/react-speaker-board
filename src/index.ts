import ReactSpeakerBoard from "./components/ReactSpeakerBoard";
import Title from "./components/Title";
import Text from "./components/Text";
import Content from "./components/Content";
import Spacer from "./components/Spacer";
import Layout from "./components/Layout";
import Theme from "./components/Theme";
import Square from "./components/Square";
import Circle from "./components/Circle";
import TriangleUp from "./components/TriangleUp";
import TriangleRight from "./components/TriangleRight";
import TriangleDown from "./components/TriangleDown";
import TriangleLeft from "./components/TriangleLeft";
import AllSlidePdf from "./components/AllSlidePdf";

export {
  ReactSpeakerBoard,
  Title,
  Content,
  Spacer,
  Layout,
  Theme,
  Square,
  Circle,
  TriangleUp,
  TriangleRight,
  TriangleDown,
  TriangleLeft,
  Text,
  AllSlidePdf,
};
