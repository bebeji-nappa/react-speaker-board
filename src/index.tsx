import React from "react";
import html2canvas from 'html2canvas';
import jsPDF from 'jspdf';

import ReactSpeakerBoard from "./ReactSpeakerBoard";
import Title from "./Title";
import Text from "./Text";
import Content from "./Content";
import Spacer from "./Spacer";
import Layout from "./Layout";
import Theme from "./Theme";
import Square from "./Square";
import Circle from "./Circle";
import TriangleUp from "./TriangleUp";
import TriangleRight from "./TriangleRight";
import TriangleDown from "./TriangleDown";
import TriangleLeft from "./TriangleLeft";
import AllSlidePdf from "./AllSlidePdf";

export const useReactSpeakerBoard = () => {
  return {
    ReactSpeakerBoard,
    Title,
    Content,
    Spacer,
    Layout,
    Theme,
    Square,
    Circle,
    TriangleUp,
    TriangleRight,
    TriangleDown,
    TriangleLeft,
    Text,
    AllSlidePdf,
  }
};

export const useSlideOutputFiles = () => {
  const download = async (slide: (() => JSX.Element)[], fileName?: string) => {
    const slideLength = slide.length;
    let pdf = new jsPDF({
      orientation: 'l',
      unit: 'px',
      format: 'a4',
    });
    for (let i = 1; i <= slideLength; i++) {
      console.log(`slide-${i}`);
      const target = document.getElementById(`slide-${i}`);
      if (target === null) return;

      const canvas = await html2canvas(target)
      const imgData = canvas.toDataURL('image/svg', 1.0);
      pdf.addImage(imgData, 'SVG', 0, 60, canvas.width / 6, canvas.height / 6, `slide-${i}`);
      if (i !== slideLength) pdf.addPage();
    }
    pdf.save(fileName ?? "reactspeakerboard.pdf");
  };
  return { download };
}
