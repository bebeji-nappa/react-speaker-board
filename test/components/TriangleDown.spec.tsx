import React from "react";
import renderer from "react-test-renderer";
import { TriangleDown } from "../../src";

describe("TriangleDown component", () => {
  it("should be correctly rendered", () => {
    const component = renderer.create(
      <TriangleDown size="80" height="40" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
})
