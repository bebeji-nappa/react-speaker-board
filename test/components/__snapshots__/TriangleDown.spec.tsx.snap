// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TriangleDown component should be correctly rendered 1`] = `
<div
  className="triangle"
  style={
    {
      "borderLeft": "80px solid transparent",
      "borderRight": "80px solid transparent",
      "borderTop": "40px solid black",
      "height": "0",
      "width": "0",
    }
  }
/>
`;
